caster.go

```diff
@@ -0,0 +1,41 @@
+package qkit
+
+import "encoding/json"
+
+// Converts any type to a given type. If conversion fails, it returns the zero value of the given type.
+func Cast[T any](val any) T {
+	if val, ok := val.(T); ok {
+		return val
+	}
+	var zero T
+	return zero
+}
+
+// Converts any type to a given struct type.
+func ToStruct[T any](m any) T {
+	var value T
+	bytes, _ := json.Marshal(m)
+	json.Unmarshal(bytes, &value)
+	return value
+}
+
+// Converts any type to a map[string]interface{}.
+func ToMap(s any) map[string]interface{} {
+	m := make(map[string]interface{})
+	bytes, _ := json.Marshal(s)
+	json.Unmarshal(bytes, &m)
+	return m
+}
+
+// Generates a byte array from any type.
+func ToBytes(v interface{}) []byte {
```

Comment:
ToBytes is too generic IMO, json.Marshal already accepts any. 
If looking for shortcut function maybe ToJSON and FromJSON make more sense.

also we already have `Ok` and `Must` wrapper functions which let you shorten the error handling like:
1. Ok(json.Marshal(...)) --> []byte, on error returns nil
2. Must(json.Marshal(...)) --> []byte, on error panics

Reply:
Got it, can make do with these instead, easier error handling was the primary use case and these should suffice